src/main/java/com/tg5/domain/AccountCurrency.java

```diff
@@ -0,0 +1,42 @@
+package com.tg5.domain;
+
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import lombok.Data;
+
+import java.math.BigDecimal;
+
+
+@Embeddable
+@Data
+public  class AccountCurrency {
+
+    AccountCurrency(BigDecimal balance, MoneyType moneyType) {
+        this.balance = balance;
+        this.moneyType = moneyType;
+    }
+
+    public AccountCurrency() {
+
+    }
+
+    public enum MoneyType {VirtualDollar, Points}
+    private  BigDecimal balance;
+
+    @Enumerated(EnumType.STRING)
+    private  MoneyType moneyType;
+
+
+    public  AccountCurrency plus(AccountCurrency currency){
```

Comment:
If this class not meant to be used directly, make this abstract?

Reply:
that was a previous idea, but unable to make it hibernate recognized when used as a type in other classes,  I reverted to a concrete type. I haven't figured out how to make abstract work with hibernate